### ISSUE TITLE
Renderer Error

### ISSUE THREAD

**Issue by gentlegiantJGC** [Khroki/MCEdit-Unified#808]
Running current code from github from source.
Copying a selection / cloning a selection of the world causes the following traceback error in the console and the copied selection does not render in the clipboard

Copying a `33*32*1` selection
```
Traceback (most recent call last):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3897, in workOnChunk
    for _ in faceInfoCalculator:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3879, in calcFacesForChunkRenderer
    for _ in calc:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 389, in calcFaces
    for _ in self.renderer.chunkCalculator.calcFacesForChunkRenderer(self):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 858, in calcFacesForChunkRenderer
    for _ in self.calcHighDetailFaces(cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1075, in calcHighDetailFaces
    for _ in self.computeGeometry(chunk, areaBlockMats, facingBlockIndices, areaBlockLights, cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1123, in computeGeometry
    chunkRenderer):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1144, in computeCubeGeometry
    texMap):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1653, in makeGenericVertices
    vertexArray = self.makeTemplate(direction, blockIndices)
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1151, in makeTemplate
    return self.precomputedVertices[direction][blockIndices]
IndexError: boolean index did not match indexed array along dimension 2; dimension is 16 but corresponding boolean dimension is 1
Traceback (most recent call last):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3897, in workOnChunk
    for _ in faceInfoCalculator:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3879, in calcFacesForChunkRenderer
    for _ in calc:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 389, in calcFaces
    for _ in self.renderer.chunkCalculator.calcFacesForChunkRenderer(self):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 858, in calcFacesForChunkRenderer
    for _ in self.calcHighDetailFaces(cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1075, in calcHighDetailFaces
    for _ in self.computeGeometry(chunk, areaBlockMats, facingBlockIndices, areaBlockLights, cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1123, in computeGeometry
    chunkRenderer):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1144, in computeCubeGeometry
    texMap):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1653, in makeGenericVertices
    vertexArray = self.makeTemplate(direction, blockIndices)
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1151, in makeTemplate
    return self.precomputedVertices[direction][blockIndices]
IndexError: boolean index did not match indexed array along dimension 0; dimension is 16 but corresponding boolean dimension is 1
Traceback (most recent call last):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3897, in workOnChunk
    for _ in faceInfoCalculator:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3879, in calcFacesForChunkRenderer
    for _ in calc:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 389, in calcFaces
    for _ in self.renderer.chunkCalculator.calcFacesForChunkRenderer(self):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 858, in calcFacesForChunkRenderer
    for _ in self.calcHighDetailFaces(cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1075, in calcHighDetailFaces
    for _ in self.computeGeometry(chunk, areaBlockMats, facingBlockIndices, areaBlockLights, cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1123, in computeGeometry
    chunkRenderer):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1144, in computeCubeGeometry
    texMap):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1653, in makeGenericVertices
    vertexArray = self.makeTemplate(direction, blockIndices)
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1151, in makeTemplate
    return self.precomputedVertices[direction][blockIndices]
IndexError: boolean index did not match indexed array along dimension 2; dimension is 16 but corresponding boolean dimension is 1
Traceback (most recent call last):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3897, in workOnChunk
    for _ in faceInfoCalculator:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3879, in calcFacesForChunkRenderer
    for _ in calc:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 389, in calcFaces
    for _ in self.renderer.chunkCalculator.calcFacesForChunkRenderer(self):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 858, in calcFacesForChunkRenderer
    for _ in self.calcHighDetailFaces(cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1075, in calcHighDetailFaces
    for _ in self.computeGeometry(chunk, areaBlockMats, facingBlockIndices, areaBlockLights, cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1123, in computeGeometry
    chunkRenderer):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1144, in computeCubeGeometry
    texMap):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1653, in makeGenericVertices
    vertexArray = self.makeTemplate(direction, blockIndices)
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1151, in makeTemplate
    return self.precomputedVertices[direction][blockIndices]
IndexError: boolean index did not match indexed array along dimension 2; dimension is 16 but corresponding boolean dimension is 1
Traceback (most recent call last):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3897, in workOnChunk
    for _ in faceInfoCalculator:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3879, in calcFacesForChunkRenderer
    for _ in calc:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 389, in calcFaces
    for _ in self.renderer.chunkCalculator.calcFacesForChunkRenderer(self):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 858, in calcFacesForChunkRenderer
    for _ in self.calcHighDetailFaces(cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1075, in calcHighDetailFaces
    for _ in self.computeGeometry(chunk, areaBlockMats, facingBlockIndices, areaBlockLights, cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1123, in computeGeometry
    chunkRenderer):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1144, in computeCubeGeometry
    texMap):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1653, in makeGenericVertices
    vertexArray = self.makeTemplate(direction, blockIndices)
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1151, in makeTemplate
    return self.precomputedVertices[direction][blockIndices]
IndexError: boolean index did not match indexed array along dimension 2; dimension is 16 but corresponding boolean dimension is 1
Traceback (most recent call last):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3897, in workOnChunk
    for _ in faceInfoCalculator:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3879, in calcFacesForChunkRenderer
    for _ in calc:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 389, in calcFaces
    for _ in self.renderer.chunkCalculator.calcFacesForChunkRenderer(self):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 858, in calcFacesForChunkRenderer
    for _ in self.calcHighDetailFaces(cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1075, in calcHighDetailFaces
    for _ in self.computeGeometry(chunk, areaBlockMats, facingBlockIndices, areaBlockLights, cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1123, in computeGeometry
    chunkRenderer):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1144, in computeCubeGeometry
    texMap):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1653, in makeGenericVertices
    vertexArray = self.makeTemplate(direction, blockIndices)
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1151, in makeTemplate
    return self.precomputedVertices[direction][blockIndices]
IndexError: boolean index did not match indexed array along dimension 0; dimension is 16 but corresponding boolean dimension is 1
```

Copying a `4*3*1` selection
```
Traceback (most recent call last):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3897, in workOnChunk
    for _ in faceInfoCalculator:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3879, in calcFacesForChunkRenderer
    for _ in calc:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 389, in calcFaces
    for _ in self.renderer.chunkCalculator.calcFacesForChunkRenderer(self):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 858, in calcFacesForChunkRenderer
    for _ in self.calcHighDetailFaces(cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1075, in calcHighDetailFaces
    for _ in self.computeGeometry(chunk, areaBlockMats, facingBlockIndices, areaBlockLights, cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1123, in computeGeometry
    chunkRenderer):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1144, in computeCubeGeometry
    texMap):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1653, in makeGenericVertices
    vertexArray = self.makeTemplate(direction, blockIndices)
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1151, in makeTemplate
    return self.precomputedVertices[direction][blockIndices]
IndexError: boolean index did not match indexed array along dimension 0; dimension is 16 but corresponding boolean dimension is 4
Traceback (most recent call last):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3897, in workOnChunk
    for _ in faceInfoCalculator:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3879, in calcFacesForChunkRenderer
    for _ in calc:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 389, in calcFaces
    for _ in self.renderer.chunkCalculator.calcFacesForChunkRenderer(self):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 858, in calcFacesForChunkRenderer
    for _ in self.calcHighDetailFaces(cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1075, in calcHighDetailFaces
    for _ in self.computeGeometry(chunk, areaBlockMats, facingBlockIndices, areaBlockLights, cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1123, in computeGeometry
    chunkRenderer):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1144, in computeCubeGeometry
    texMap):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1653, in makeGenericVertices
    vertexArray = self.makeTemplate(direction, blockIndices)
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1151, in makeTemplate
    return self.precomputedVertices[direction][blockIndices]
IndexError: boolean index did not match indexed array along dimension 0; dimension is 16 but corresponding boolean dimension is 4
Destination:  _Vector(x=124L, y=4L, z=-38L)
Traceback (most recent call last):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3897, in workOnChunk
    for _ in faceInfoCalculator:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 3879, in calcFacesForChunkRenderer
    for _ in calc:
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 389, in calcFaces
    for _ in self.renderer.chunkCalculator.calcFacesForChunkRenderer(self):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 858, in calcFacesForChunkRenderer
    for _ in self.calcHighDetailFaces(cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1075, in calcHighDetailFaces
    for _ in self.computeGeometry(chunk, areaBlockMats, facingBlockIndices, areaBlockLights, cr, blockRenderers):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1123, in computeGeometry
    chunkRenderer):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1144, in computeCubeGeometry
    texMap):
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1653, in makeGenericVertices
    vertexArray = self.makeTemplate(direction, blockIndices)
  File "C:\Users\james_000\Desktop\New folder (4)\MCEdit-Unified-master\renderer.py", line 1151, in makeTemplate
    return self.precomputedVertices[direction][blockIndices]
IndexError: boolean index did not match indexed array along dimension 0; dimension is 16 but corresponding boolean dimension is 4
```

**Comment by gentlegiantJGC** [comment]
Additional information: When a whole semi-chunk is selected this will render as intended however anything in the selection beyond this will not until the region selects the whole of the next chunk.

**Comment by Podshot** [comment]
Interesting, I can't seem to reproduce this. Are there any modded blocks or blocks that have special models in MCEdit (like End Rods, Levers) in the selection? Would you be able to send us a copy of the world along with the coordinates of the selection box where this error occurs?

**Comment by gentlegiantJGC** [comment]
This is independent of the world loaded. I generated a super flat world in MCedit and did the same thing and got the same results.
Exactly two semi-chunks selected - works as intended
![2017-06-19 11-24-00-am](https://user-images.githubusercontent.com/11940194/27280970-dc50778c-54e1-11e7-9f74-80fd5c1a1754.png)
One layer removed from one semi-chunk and that chunk no longer renders
![2017-06-19 11-24-07-am](https://user-images.githubusercontent.com/11940194/27280971-dc517bbe-54e1-11e7-856c-81e96d7f2294.png)
top layer removed from both semi-chunks and they both stop rendering
![2017-06-19 11-24-20-am](https://user-images.githubusercontent.com/11940194/27280969-dc4f6bee-54e1-11e7-95d3-768b7621ba26.png)

**Comment by gentlegiantJGC** [comment]
done a bit more digging into this problem.

line 1151 in renderer.py
`return self.precomputedVertices[direction][blockIndices]`

self.precomputedVertices[direction] always seems to have a length of 16 (I don't really understand the format of this but it is probably coordinates.)
blockIndices is multiple nested lists containing true or false. Below taken from a direction=0 of a 3W x 2L x 1H region. the length of blockIndices seems to be the width, len(blockIndices[n]) the length and len(blockIndices[n][m]) the height. 
```
[[[False]
  [False]]

 [[False]
  [False]]

 [[ True]
  [ True]]]
```

the issue seems to be that one of the two lists is the wrong dimension and as throwing a dimension missmatch as a result. This probably explains why the issue doesn't exist when the selection is alligned to and selects a whole chunk because self.precomputedVertices is always a 16*16*16

**Comment by gentlegiantJGC** [comment]
@neonerz you are running source on windows right? Have you encountered this issue when cloning a selection that doesn't fill a whole sub-chunk?

**Comment by neonerz** [comment]
I've seen that error that you mentioned:

> IndexError: boolean index did not match indexed array along dimension 0; dimension is 16 but corresponding boolean dimension is 4

I don't do a whole lot of cloning/copying, so I might not be the best person to speak on the issue, but even though I've seen that error, I never had an issue with cloning and copying. I do do quite a bit importing of schematics though, and I never had the issue there.

Does the issue only happen when you move it to a different subchunk within the same chunk? It could be why I've never seen the issue. 

I always assumed that error I was getting was related to the version of numpy I'm using in windows. I forget why I was under that impression though.

**Comment by neonerz** [comment]
Oh, I might have been misunderstanding your problem. Do you mean it doesn't visually render while it's still a selection, but if you actually paste it, it's OK? If so, I have seen that before, I just chalked it up to a MCEdit render bug. It never really effected my work flow, so I never really thought too hard about it. It also doesn't seem to always happen to me (though, like I said, I never paid too much mind to it)

**Comment by gentlegiantJGC** [comment]
The issue arises when a sub-chunk is partially selected and copied. It is a rendering issue so the actual copying operation will work as normal but the partially selected sub-chunks will not render while the selection can be moved around (when it is surrounded by the green box before actually pasted into the world. I don't know how to better explain that)

It may be that every time you are doing it you are selecting whole chunks so this issue doesn't arise

**Comment by gentlegiantJGC** [comment]
Just seen your other message. Yes to all of that.
I am mostly just trying to work out if it is a Windows based problem or a general problem

**Comment by gentlegiantJGC** [comment]
@neonerz Did I also see you mentioned running a Linux VM?

Steps to reproduce
1) Select one block
2) click clone
3) view log

**Comment by naor2013** [comment]
Ok, so I only saw the error you shared in the first comment and it seems like the error there doesn't have anything to do with any renderer issue. I can't reproduce any renderer issue but if you still exprience it, please reopen this issue and tell us how to get it

**Comment by gentlegiantJGC** [comment]
Out of interest what OS are you running on?

Steps to reproduce are as follows. I am running on windows from source.

1) Select one block
2) click clone
3) view log

The issue is a dimension mismatch in renderer.py at line 1151 as in the original error posted.

return self.precomputedVertices[direction][blockIndices]

self.precomputedVertices always has a dimension of 16x16x16. blockIndices has dimensions equal to the box selected.

This is where the original error comes from

"dimension is 16 but corresponding boolean dimension is 4"
(4 being the size of the selection)

With a bit more information of the normal dimensions of each of these variables I could do some more debugging.

This issue does not affect sub-chunks that are fully selected since blockIndices has a dimension of 16 and thus there isn't a dimension mismatch

**Comment by naor2013** [comment]
Seems like I didn't fix the issue, just caused a bigger one... I have an issue with the Clone in general, maybe because of an outdated library or something so I can't reproduce the issue myself

**Comment by gentlegiantJGC** [comment]
I will not be able to test any changes for about another week but after that I can debug anything.
My guess is that blockIndices needs to be padded out so that it is the same dimension as what it is being looked up in.

Some info on what the variables are and what is in them would help. There isn't much documentation in the program

**Comment by naor2013** [comment]
The issue seems to be in FakeLevel. FakeLevel seems to be something it creates to show the cloned part but it creates it in the dimension of the cloned area which causes issue if they don't fit with the chunk borders of the real level. Not sure if I'm right about this but I'm pretty sure it's that. We'll keep trying to fix it in the next few days, hopefully it will work by the time you'll be able to test.

**Comment by Podshot** [comment]
Haven't been able to reproduce the non-rendering error, but I have encountered the error message. I'm investigating whether this is caused by a NumPy version or our own code

**Comment by Podshot** [comment]
@gentlegiantJGC What version of NumPy do you have installed?

**Comment by gentlegiantJGC** [comment]
appdirs (1.4.3)
Cython (0.25.2)
ftputil (3.3.1)
numpy (1.13.0)
olefile (0.44)
Pillow (4.1.1)
pip (9.0.1)
protobuf (3.3.0)
pygame (1.9.3)
pygubu (0.9.8.1)
PyOpenGL (3.1.0)
pypiwin32 (219)
PyYAML (3.12)
setuptools (28.8.0)
Shapely (1.6.1)
six (1.10.0)

That is the list of everything I have installed. Not all of them are MCedit related and I can't remember which ones are

**Comment by gentlegiantJGC** [comment]
I have also had another issue very similar to [this](https://github.com/Khroki/MCEdit-Unified/pull/813) when pasting which may also be numpy related but I haven't had a chance to debug it

**Comment by gentlegiantJGC** [comment]
@Podshot I have just uninstalled that version and installed version 1.9.3 as the readme says and that seems to have fixed both the rendering issues and the issue I just mentioned

**Comment by Podshot** [comment]
Ok, thanks for the confirmation. It seems numpy <= 1.13 causes the renderer issue, anything lower just gives a warning message, and 1.9.3 removes the message completely, but from what I've tested the rotating error still occurs on 1.9.3. I'll talk with the team and discuss how we want to solve this issue

**Comment by naor2013** [comment]
Finally fixed, it was a change in the Numpy library in version 1.13 like Podshot said, but now it works as intended :D
